src/main/java/frc/trigon/robot/utilities/mechanisms/DoubleJointedArmMechanism2d.java

```diff
@@ -0,0 +1,219 @@
+package frc.trigon.robot.utilities.mechanisms;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import org.littletonrobotics.junction.Logger;
+
+/**
+ * A Mechanism2d object to display the current angle and the target angle of a double jointed arm.
+ */
+public class DoubleJointedArmMechanism2d {
+    private final String key;
+    private final Mechanism2d mechanism;
+    private final MechanismLigament2d
+            currentPositionFirstLigament,
+            currentPositionSecondLigament,
+            targetPositionFirstLigament,
+            targetPositionSecondLigament;
+
+    public DoubleJointedArmMechanism2d(String key) {
+        this(key, MechanismConstants.BLUE);
+    }
+
+    private DoubleJointedArmMechanism2d(String key, Color8Bit mechanismColor) {
```

Comment:
Private?
There should probably be different colors for each mech
Input each mech's length
make sure that its whole mechanism size isn't exactly the length of both arms; give some wiggle room as a 10% constant or something. So the mechanism will be 10% longer than both arms together * 2.